bin/Debugger.Sample/DebugService.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "DebugProtocolHandler.h"
+
+#include <ChakraDebugService.h>
+#include <exception>
+#include <string>
+
+class DebugService
+{
+private:
+    JsDebugService m_service{ nullptr };
+
+public:
+    DebugService()
+    {
+        JsDebugService service;
+
+        JsErrorCode result = JsDebugServiceCreate(&service);
+
+        if (result != JsNoError)
+        {
+            throw new std::exception("Unable to create debug service.");
+        }
+
+        m_service = service;
+    }
+
+    ~DebugService()
+    {
+        Destroy();
+    }
+
+    JsErrorCode Close()
+    {
+        JsErrorCode result = JsDebugServiceClose(m_service);
+
+        return result;
+    }
+
+    JsErrorCode Destroy()
+    {
+        JsErrorCode result = JsNoError;
+
+        if (m_service != nullptr)
+        {
+            result = JsDebugServiceDestroy(m_service);
+
+            if (result == JsNoError)
+            {
+                m_service = nullptr;
+            }
+        }
+
+        return result;
+    }
+
+    JsErrorCode Listen(uint16_t port)
+    {
+        JsErrorCode result = JsDebugServiceListen(m_service, port);
+
+        return result;
+    }
+
+    JsErrorCode RegisterHandler(std::string const& runtimeName, DebugProtocolHandler& protocolHandler, bool breakOnNextLine)
```

Comment:
minor, but wouldn't it be better to put the protocolHandler argument first?

Reply:
Since the underlying API call has them in this order I think it makes sense to keep it as-is for now.  There's probably an argument to be made for the other order, but I don't think it's worth taking a breaking change to the API right now.